### DISCUSSION TITLE
Allow custom user-agent setting

### DISCUSSION THREAD

**Discussion by mysterfr** [pymedusa/Medusa#11057] [Ideas & Feature requests]
Hi,

It seems that pyMedusa advertises an explicit user-agent string, for instance :
`Medusa/1.0.9 (Linux; 5.10.0-19-amd64; 0856dfab-794f-11ed-8a1b-0242ac110004)`

I witnessed that certain websited, protect by cloudflare, systematically block the traffic because of it.
It's the case for instance on YGGtorrent.

My pyMedusa instance used to work for quite a while, both directly but also through Jackett+Flaresolver.
For a couple of weeks now, none of them work.

When I change my browser's user-agent to the exact same string and try to browse the website, it get stuck at the CloudFlare security check, and even though as a human I solve the check, it loops back to blocking me and proving again that I'm human.

Switching back the user-agent to Firefox', all goes well again.

So to me it seems like Cloudflare purposedly blocks user-agent with the "Medusa" string in it, even though the security check is successfully completed by a human.

I would like to be able to override the default Medusa user-agent string with a custom one. Would be even better if this override could be done on a per-provider basis instead of globally.

Thanks !

**Comment by p0psicles** [comment]
We're not adding options to set a custom user agent. But we could set a specific user agent for yggtorrent. You could also do a pull request your self

**Reply by mysterfr** [reply]
Thanks @p0psicles , I'll have a look at that.
